service/handlers.go

```diff
@@ -117,22 +117,32 @@ func SignHandler(w http.ResponseWriter, r *http.Request) ErrorResponse {
 		return ErrorNilData
 	}
 
-	// Read the full request body
-	data, err := ioutil.ReadAll(r.Body)
+	defer r.Body.Close()
+
+	// Use snapd assertion module to decode the assertions in the request stream
+	dec := asserts.NewDecoder(r.Body)
+	assertion, err := dec.Decode()
+	if err == io.EOF {
+		return ErrorEmptyData
+	}
```

Comment:
maybe these two error could be evaluated together as io.EOF is also a err != nil. Actually if err == io.EOF nothing is logged using logMessage

Reply:
yes, in the changes I dropped by mistake:
logMessage("SIGN", "invalid-assertion", "No data supplied for signing")

it seems putting that back instead of unifying the two cases would be the more backward compatible approach